### DISCUSSION TITLE
Navigating to the parent directory via DirectoryTree

### DISCUSSION THREAD

**Discussion by anzhi0708** [Textualize/textual#4422] [Q&A]
The `DirectoryTree` widget seems to not have a method for moving to the parent directory (only child items under current directory). What should I do?

**Comment by TomJGooding** [comment]
It's not really clear exactly what you're asking here, but here's a quick example:which might help clarify what you're looking for:

```python
from pathlib import Path

from textual import on
from textual.app import App, ComposeResult
from textual.widgets import Button, DirectoryTree


class DirectoryTreeApp(App):
    def compose(self) -> ComposeResult:
        yield Button("Navigate to parent directory")
        yield DirectoryTree("./")

    @on(Button.Pressed)
    def update_directory_tree(self) -> None:
        tree = self.query_one(DirectoryTree)
        assert isinstance(tree.path, Path)
        tree.path = tree.path.parent.resolve()
        tree.reload()


if __name__ == "__main__":
    app = DirectoryTreeApp()
    app.run()
```

**Comment by palao** [comment]
Thank you for the solution.
It would be good to have something built-in in the widget though, wouldn't it?

**Reply by mlopezgva** [reply]
Hi!

Came here for the same reason... The goal of a directory tree browser is to navigate through it... And going to the parent directory I think is a must (configurable, something like: `show_parent_dir: Bool = False`, or even `show_hidden: Bool = False`).

It would be really nice. This is my 2nd incursion in Python and my first with Textual. Right now I have set the "Back" key event to "go to parent", but it feels like unnecessary to have do that. :-)

**Reply by edward-jazzhands** [reply]
I agree that this should be built in to the widget. My opinion is that someone should do up a PR draft to add a method for this.

**Comment by pycaw** [comment]
My solution; if you click on the root node on the top, it will switch to parent if it can:
```py
from pathlib import Path
from textual.widgets import DirectoryTree, Tree
from textual.widgets.directory_tree import DirEntry

class MyDirectoryTree(DirectoryTree):
    async def on_tree_node_collapsed(self, event: Tree.NodeExpanded[DirEntry]):
        event.stop()
        dir_entry = event.node.data
        if dir_entry is None or not event.node.is_root:
            return
        assert isinstance(self.path, Path)
        parent = self.path.resolve().parent
        if self._safe_is_dir(parent):
            self.path = parent
            await self.reload()
```

Does what is expected. Root directory name will be displayed once you use the feature.

If you wanted to select a directory, you should exclude root:
```py
    def on_directory_tree_directory_selected(self, event: DirectoryTree.DirectorySelected):
        if not event.node.is_root:
            self.dismiss(event.path)
```
